cfme/intelligence/reports/menus.py

```diff
@@ -180,6 +180,30 @@ def manage_folder(self, group, folder=None):
             view.manager.commit()
             view.save_button.click()
 
+    def move_reports(self, group, folder, subfolder, reports):
+        """ Moves a list of reports to a given menu
+        Args:
+            group: User group
+            folder: Parent of the subfolder where reports are to be moved.
+            subfolder: Subfolder under which the reports are to be moved.
+            reports: List of reports that are to be moved.
+        """
+        if not isinstance(reports, list):
+            reports = [reports]
+
+        with self.manage_subfolder(group, folder, subfolder) as selected_menu:
+            selected_options = selected_menu.parent_view.report_select.all_options
+            for report in reports:
+                if report in selected_options:
+                    # pass, since the report is already present
+                    pass
```

Comment:
I think you took my suggestion in the wrong way :)

```
            for report in reports:
                if report in selected_options:
                    # pass, since the report is already present
                    pass
```
what is the use of this for loop?

Reply:
I understand your question now :sweat_smile: 
I think the new changes will now make a sense of why I am using the loop. If the report is already present, we stay on the same page. To get out of that page, we have to click cancel. I had added a new view(`AllReportMenusView) and navigator which were not present earlier and modified the `move_reports` method.

Reply:
Let me know as PRT cleared... LGTM :+1: